internal/users/service/auth_service.go

```diff
@@ -0,0 +1,87 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/KnoblauchPilze/user-service/pkg/communication"
+	"github.com/KnoblauchPilze/user-service/pkg/db"
+	"github.com/KnoblauchPilze/user-service/pkg/errors"
+	"github.com/KnoblauchPilze/user-service/pkg/repositories"
+	"github.com/google/uuid"
+)
+
+type AuthService interface {
+	Authenticate(ctx context.Context, apiKey uuid.UUID) (communication.AuthorizationDtoResponse, error)
+}
+
+type authServiceImpl struct {
+	conn       db.ConnectionPool
+	aclRepo    repositories.AclRepository
+	apiKeyRepo repositories.ApiKeyRepository
+	userLimit  repositories.UserLimitRepository
+}
+
+func NewAuthService(conn db.ConnectionPool, repos repositories.Repositories) AuthService {
+	return &authServiceImpl{
+		conn:       conn,
+		aclRepo:    repos.Acl,
+		apiKeyRepo: repos.ApiKey,
+		userLimit:  repos.UserLimit,
+	}
+}
+
+func (s *authServiceImpl) Authenticate(ctx context.Context, apiKey uuid.UUID) (communication.AuthorizationDtoResponse, error) {
```

Comment:
As advertised:
* we check if the API key exists in our system
* if yes we retrieve the user to which this API key belongs
* we fetch the ACLs for this user
* we fetch the limits for this user
* we return everything in a single JSON object